server/index.js

```diff
@@ -50,6 +50,110 @@ app.get('/api/products/:id', (req, res, next) => {
     .catch(err => next(err));
 });
 
+app.get('/api/cart', (req, res, next) => {
+  if (!req.session) {
+    return (
+      res.status(400).json({
+        error: 'cart is empty'
+      })
+    );
+  }
```

Comment:
Since we are using some `session` middleware, the `session` will be an object, which is always truthy. Instead, you should be checking for the existence of a `cartId` in the `req.session`.

Moreover, this is not a failure scenario. If there is no `cartId` in the `session`, that's ok, that just means the client's cart is empty. So they should receive an empty array instead of an error message.